kernel_gateway/services/notebooks/handlers.py

```diff
@@ -29,82 +34,86 @@ def _request_assignment_for_lang(self, kernel_name, expression):
             statement = self._assignment_statements[None]
         return statement.format(expression)
 
-    def _process_messages(self, parent_header):
-        idle = False
-        execute_result = None
-        stream_messages = []
-        error_message = None
-        while(not idle):
-            iopub_message = self.kernel_client.get_iopub_msg(block=True, timeout=self.execution_timeout)
-            # Only look at messages which are derived from the parent_header
-            if iopub_message['parent_header']['msg_id'] == parent_header:
-                # On idle status, exit our loop
-                if iopub_message['header']['msg_type'] == 'status' and iopub_message['content']['execution_state'] == 'idle':
-                    idle = True
-                # Store the execute result
-                elif iopub_message['header']['msg_type'] == 'execute_result':
-                    execute_result = iopub_message['content']['data']
-                # Accumulate the stream messages
-                elif iopub_message['header']['msg_type'] == 'stream':
-                    stream_messages.append(iopub_message['content']['text'])
-                # Store the error message
-                elif iopub_message['header']['msg_type'] == 'error':
-                    error_message = 'Error {}: {} \n'.format(
-                        iopub_message['content']['ename'],
-                        iopub_message['content']['evalue']
-                    )
-
-        if error_message is not None:
-            return True, error_message
-        elif execute_result is not None and execute_result is not '':
-            return False, execute_result
-        else:
-            return False, ''.join(stream_messages)
-
-    def _send_code(self, code):
-        # execute the code and return the result and HTTP status code
-        parent_header = self.kernel_client.execute(code)
-        status = 200
-        error, result = self._process_messages(parent_header=parent_header)
-
-        if error:
-            status = 500
-
-        return result, status
-
+    def on_recv(self, msg):
```

Comment:
Would be good to have a docstring on this one explaining what it's doing overall.